ui/pages/ApplicationAdd.tsx

```diff
@@ -163,6 +164,14 @@ function AddApplication({ className }: Props) {
     setAuthOpen(true);
   };
 
+  React.useEffect(() => {
+    if (!formState.provider) {
+      return;
+    }
+
+    check(formState.provider);
+  }, [formState.provider, authSuccess]);
```

Comment:
Why do we need to check when authSuccess changes?

Reply:
To get the `isAuthenticated` to flip from `false` or `null` to `true`. Makes everything slightly cleaner.